### PR TITLE
TS-11: Implement help pop up window

### LINKED ISSUE #11: [TSUI] Implement help pop up window

### PR DESCRIPTION
**Pull request by ihordubas**
#11

### CODE REVIEW

File: src/components/common/chart/chart.sass [hunk 1]
```diff
@@ -30,6 +30,8 @@
         .chart-subtitle
             color: $yellow
             font-weight: bold
+        .info-icon
```
Comment (PetrushynskyiOleksii) [inline src/components/common/chart/chart.sass]: let's call it chart-info-icon

File: src/components/common/chart/chartHeader.jsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
 import React from "react"
 
+import Icon from "@icons/icon"
+import { NORMAL_ICON_SIZE } from "@utils/constants"
```
Comment (PetrushynskyiOleksii) [inline src/components/common/chart/chartHeader.jsx:4]: add empty line after

File: src/components/common/chart/chartHeader.jsx [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
 import "./chart.sass"
 
 
-const ChartHeader = ({ title, subtitle, refresh }) => {
+const ChartHeader = ({ title, subtitle, refresh, helpPopupOpen, }) => {
```
Comment (PetrushynskyiOleksii) [inline src/components/common/chart/chartHeader.jsx]: let's rename helpPopupOpen => openHelp everywhere

File: src/components/helpPopup/helpPopup.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import React from "react"
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: let's move it to components/chart and call it chartHelp

File: src/components/helpPopup/helpPopup.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+import Icon from "@icons/icon"
+import { NORMAL_ICON_SIZE } from "@utils/constants"
+
+import "./helpPopup.sass"
+
+
+const HelpPopup = ({ open, close, route, currentChartName }) => {
+
+    const CHART_DESCRIPTION = {
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: move it out of component

File: src/components/helpPopup/helpPopup.jsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import Dialog from "@material-ui/core/Dialog"
+import DialogContent from "@material-ui/core/DialogContent"
+
+import Icon from "@icons/icon"
+import { NORMAL_ICON_SIZE } from "@utils/constants"
+
+import "./helpPopup.sass"
+
+
+const HelpPopup = ({ open, close, route, currentChartName }) => {
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: let's rename currentChartName -> chartName, HelpPopup -> ChartHelp

File: src/components/helpPopup/helpPopup.jsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        "Сихівський": "10 description of chart",
+        "Личаківський": "11 description of chart",
+        "Залізничний": "12 description of chart",
+        "Шевченківський": "13 description of chart",
+        "Франківський": "14 description of chart",
+    }
+
+    return (
+        <Dialog open={open} onClose={close}>
+            <DialogContent className="popup">
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: chart-help

File: src/components/helpPopup/helpPopup.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+                <div className="popup-header">
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: chart-help-header

File: src/components/helpPopup/helpPopup.jsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                    <div className="title">{currentChartName}</div>
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: chart-help-title

File: src/components/helpPopup/helpPopup.jsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+                    <div className="transport-number">{route}</div>
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: chart-help-route

File: src/components/helpPopup/helpPopup.jsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+                </div>
+                <div className="popup-text">{CHART_DESCRIPTION[currentChartName]}</div>
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: chart-help-desc

File: src/components/helpPopup/helpPopup.jsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+                <div className="download-button">
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: chart-download-button

File: src/components/helpPopup/helpPopup.jsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+                    <Icon name="download-icon" className="download-icon" size={NORMAL_ICON_SIZE}/>
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.jsx]: chart-download-icon

File: src/components/helpPopup/helpPopup.sass [hunk 14]
```diff
@@ -0,0 +1,40 @@
+.popup
```
Comment (PetrushynskyiOleksii) [inline src/components/helpPopup/helpPopup.sass]: rename all of them based on comments above

File: src/components/pages/districts.jsx [hunk 15]
```diff
@@ -1,24 +1,39 @@
...
+import React, { useEffect, useState } from "react"
 
 import ChartsContainer from "@components/common/chart/chartsContainer"
 import DonutHistory from "@components/tiles/donutHistory/donutHistory"
+import HelpPopup from "@components/helpPopup/helpPopup"
 
 
 const DistrictsPage = () => {
+
+    const [helpPopupOpen, setHelpPopupOpen] = useState(false)
```
Comment (PetrushynskyiOleksii) [inline src/components/pages/districts.jsx]: openHelp, setOpenHelp

File: src/components/pages/districts.jsx [hunk 16]
```diff
@@ -1,24 +1,39 @@
...
+    const [currentChartName, setCurrentChartName] = useState('')
```
Comment (PetrushynskyiOleksii) [inline src/components/pages/districts.jsx]: double quotes

Comment (PetrushynskyiOleksii) [reply src/components/pages/districts.jsx]: helpChartName, setHelpChartName

File: src/components/pages/traffic.jsx [hunk 17]
```diff
@@ -12,6 +13,8 @@ import { BIG_ICON_SIZE, ROUTE_KEY, PERIOD_KEY, HOUR_SECONDS } from "@utils/const
 const TrafficPage = () => {
 
     const [searchOpen, setSearchOpen] = useState(false)
+    const [helpPopupOpen, setHelpPopupOpen] = useState(false)
+    const [currentChartName, setCurrentChartName] = useState('')
```
Comment (PetrushynskyiOleksii) [inline src/components/pages/traffic.jsx]: double quoutes

**Review by PetrushynskyiOleksii — CHANGES_REQUESTED** [review]

File: src/components/common/chart/chart.sass (hunk 1 above)
Comment (ihordubas) [reply src/components/common/chart/chart.sass]: done

File: src/components/common/chart/chartHeader.jsx (hunk 2 above)
Comment (ihordubas) [reply src/components/common/chart/chartHeader.jsx:4]: done

File: src/components/common/chart/chartHeader.jsx (hunk 3 above)
Comment (ihordubas) [reply src/components/common/chart/chartHeader.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 4 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 6 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 5 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 7 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 8 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 9 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 10 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 11 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 12 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.jsx (hunk 13 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.jsx]: done

File: src/components/helpPopup/helpPopup.sass (hunk 14 above)
Comment (ihordubas) [reply src/components/helpPopup/helpPopup.sass]: done

File: src/components/pages/districts.jsx (hunk 16 above)
Comment (ihordubas) [reply src/components/pages/districts.jsx]: done

File: src/components/pages/traffic.jsx (hunk 17 above)
Comment (ihordubas) [reply src/components/pages/traffic.jsx]: done

**Review by PetrushynskyiOleksii — APPROVED** [review]

**Pull request merged**